File: basex-api/src/main/scala/addexample.scala
import java.io._

/**
 * This example shows how documents can be added to databases.
 * Documentation: http://docs.basex.org/wiki/Clients
 *
 * @author BaseX Team 2005-12, BSD License
 */
object addexample {
  /**
   * Main method.
   * @param args command-line arguments
   */
  def main(args: Array[String]) {
    // create session
    val session = new BaseXClient("localhost", 1984, "admin", "admin")

    // create empty database
    session.execute("create db database")
    println(session.info)

    // define input stream
    var bais = new ByteArrayInputStream("<x>Hello World!</x>".getBytes)

    // add document
    session.add("world/world.xml", bais)
    println(session.info)

    // define input stream
    bais = new ByteArrayInputStream("<x>Hello Universe!</x>".getBytes)

    // add document
    session.add("universe.xml", bais)
    println(session.info)

    // run query on database
    println
    println(session.execute("xquery collection('database')"))

    // drop database
    session.execute("drop db database")

    // close session
    session.close
  }
}

File: basex-api/src/main/python/AddExample.py
# -*- coding: utf-8 -*-
# This example shows how new documents can be added.
#
# Documentation: https://docs.basex.org/wiki/Clients
#
# (C) BaseX Team 2005-12, BSD License

from BaseXClient import BaseXClient

# create session
session = BaseXClient.Session('localhost', 1984, 'admin', 'admin')

try:

    # create empty database
    session.execute("create db database")
    print(session.info())

    # add document
    session.add("world/World.xml", "<x>Hello World!</x>")
    print(session.info())

    # add document
    session.add("Universe.xml", "<x>Hello Universe!</x>")
    print(session.info())

    # run query on database
    print("\n" + session.execute("xquery collection('database')"))

    # drop database
    session.execute("drop db database")

finally:
    # close session
    if session:
        session.close()
